src/sentry/auth/superuser.py

```diff
@@ -53,6 +53,9 @@
 
 
 def is_active_superuser(request):
+    if getattr(request.user, 'is_system', False):
+        return True
```

Comment:
This is probably the most significant change. Since the `SuperuserMiddleware` is created before authentication classes run, it already has an anonymous user attached to it. Based on my limited insight, it doesn't seem that the sudo system is intended to allow user changes within a session (in fact it actively tries to prevent that).

Instead, when we detect a system user (which for now can only be supplied via the `SystemToken` in `TokenAuthentication`) we can easily force a super user session here, which will propagate nicely to every place where this is checked explicitly in the API.